packages/web/server/queries/complex/cosmos/distribution-params.ts

```diff
@@ -0,0 +1,22 @@
+import { RatePretty } from "@keplr-wallet/unit";
+import cachified, { CacheEntry } from "cachified";
+import { LRUCache } from "lru-cache";
+
+import { DEFAULT_LRU_OPTIONS } from "~/config/cache";
+import { queryDistributionParams } from "~/server/queries/cosmos/distribution";
+
+const distributionCache = new LRUCache<string, CacheEntry>(DEFAULT_LRU_OPTIONS);
+
+export function getDistributionParams({ chainId }: { chainId: string }) {
```

Comment:
Since we're still early, I'm wondering if we somewhat replicate what Keplr did with the Chain ID > Node Queries map. 

We can have a default for Osmosis chain ID, but perhaps it's a function that can make it easy to make node queries to other chains in the chain registry.

We could do this later or not do it at all, as I don't think we have a strong need to query other chains. Perhaps the approach we can take is to just create additional query functions for querying arbitrary chains by ID as needed.

Reply:
I agree with adding this feature early. I've made a simple function called `createNodeQuery` to help us do this. It also simplifies fetching node data by removing the need to constantly re-enter the `URL`. Please take a look at my latest commit